### PR TITLE
Make certificates tenant aware.

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description.
This allows to override the model GeneratedCertificates model, using monkey patch in order to filter the certificates by org.

### CODE REVIEW

File: eox_tenant/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 """
 Init for eox-tenant.
 """
-__version__ = '1.3.0'
+__version__ = '2.0.0'
```
Comment (felipemontoya) [inline eox_tenant/__init__.py]: We are already in V2.0.0. I dont think this is necessary for the PR. We should leave the releases as part of a separate PR.

File: eox_tenant/tenant_wise/__init__.py [hunk 2]
```diff
@@ -3,14 +3,17 @@
 ==================
 This module makes it possible to override the some platform Models using new proxy models.
 """
+from eox_tenant.edxapp_wrapper.certificates_module import get_certificates_models
 from eox_tenant.edxapp_wrapper.site_configuration_module import get_site_configuration_models
-from eox_tenant.tenant_wise.proxies import TenantSiteConfigProxy
+from eox_tenant.tenant_wise.proxies import TenantSiteConfigProxy, TenantGeneratedCertificateProxy
 
 SiteConfigurationModels = get_site_configuration_models()
+CertificatesModels = get_certificates_models()
```
Comment (felipemontoya) [inline eox_tenant/tenant_wise/__init__.py]: If we are only loading the model here to use it below, why not combine both lines into one?

File: requirements.in [hunk 3]
```diff
@@ -8,3 +8,4 @@ pycodestyle==2.4.0
 mock==2.0.0
 path.py==8.2.1
 django-crum==0.7.3
+django-fake-model==0.1.4
```
Comment (felipemontoya) [inline requirements.in]: This is only for testing right? Is this fake-model something we should refactor into other tests in the future?

File: setup.cfg [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 1.3.0
+current_version = 2.0.0
```
Comment (felipemontoya) [inline setup.cfg]: Same as the first file. No need to do this on this PR, also it is already in V2

File: eox_tenant/tenant_wise/proxies.py [hunk 5]
```diff
@@ -160,3 +163,62 @@ def set_key_to_cache(cls, key, value):
             value,
             EOX_TENANT_CACHE_KEY_TIMEOUT
         )
+
+
+class TenantCerticateManager(models.Manager):
```
Comment (felipemontoya) [inline eox_tenant/tenant_wise/proxies.py]: TenantCerticateManager typo -> TenantCertificateManager

File: eox_tenant/__init__.py (hunk 1 above)
Comment (andrey-canon) [reply eox_tenant/__init__.py]: @felipemontoya  Sorry, I forgot to change the base branch, this diff is not visible against the master branch

File: requirements.in (hunk 3 above)
Comment (andrey-canon) [reply requirements.in]: yes, this is only for testing, probably in other tests where we need a model from the platform

File: eox_tenant/tenant_wise/proxies.py [hunk 6]
```diff
@@ -160,3 +163,62 @@ def set_key_to_cache(cls, key, value):
...
+
+    def get_queryset(self):
+        """
+        Call parent method and filter the certificates by org.
+        """
+        generated_certificates = super(TenantCertificateManager, self).get_queryset()
+        excluded_courses = [
+            certificate.course_id
+            for certificate in generated_certificates
+            if certificate.course_id.org not in getattr(settings, "course_org_filter", [])
```
Comment (morenol) [inline eox_tenant/tenant_wise/proxies.py]: I think that we should create a function that returns

getattr(settings, key)

If we do that we can differentiate the times when we want to get a value for the current tenant with the times when we want to get general django setting.

Comment (felipemontoya) [reply eox_tenant/tenant_wise/proxies.py]: Is there ever a case outside of the internals of the eox-tenant library where we want to get the general django settings instead of the current tenant. I think not, but I would be happy to learn about such cases

Comment (felipemontoya) [reply eox_tenant/tenant_wise/proxies.py]: By internals of eox-tenant I mean the part where we load the current tenant in the settings object. This proxy would not be part of the internals

Comment (morenol) [reply eox_tenant/tenant_wise/proxies.py]: I think that right now there is not such case. But that change would do this more similar to the configuration_helpers used in openedx and any other projects could import that tenant_helpers module easily.

Comment (andrey-canon) [reply eox_tenant/tenant_wise/proxies.py]: @morenol  I think a goal for this plugin is to do the opposite, do the things the least similar to configuration helpers in order to avoid the issues we have with it.

Comment (felipemontoya) [reply eox_tenant/tenant_wise/proxies.py]: It is correct. An unspoken rule of design here was to go against the way site_configuration works
